### PR TITLE
Add sequencer cancellation tests

### PR DESCRIPTION
**Pull request by AlexChetverov**

### CODE REVIEW

File: test/sequential/sequencerStaking.cancel.test.ts [hunk 1]
```diff
@@ -0,0 +1,403 @@
...
+  }
+}
+
+async function createACollatorUser() {
+  const [user] = setupUsers();
+  await Sudo.asSudoFinalized(Assets.mintNative(user));
+  return user;
+}
+
+async function leaveSequencingIfAlreadySequencer(userAddr: string) {
```
Comment (goncer) [inline test/sequential/sequencerStaking.cancel.test.ts:46]: is it duplicated from some other test? 
Plz refactor it to an external class and reuse it in both tests

File: test/sequential/sequencerStaking.cancel.test.ts [hunk 2]
```diff
@@ -0,0 +1,403 @@
...
+    await Sudo.asSudoFinalized(
+      Sudo.sudoAsWithAddressString(
+        userAddr,
+        await SequencerStaking.unstake(chain as ChainName),
+      ),
+    );
+  }
+}
+
+async function createAnUpdateAndCancelIt(
```
Comment (goncer) [inline test/sequential/sequencerStaking.cancel.test.ts:72]: same

File: test/sequential/sequencerStaking.cancel.test.ts [hunk 3]
```diff
@@ -0,0 +1,403 @@
...
+      await SequencerStaking.provideSequencerStaking(BN_ZERO, "Ethereum"),
+    ),
+    Sudo.sudoAsWithAddressString(
+      preSetupSequencers.Arbitrum,
+      await SequencerStaking.provideSequencerStaking(BN_ZERO, "Arbitrum"),
+    ),
+  );
+});
+
+beforeEach(async () => {
```
Comment (goncer) [inline test/sequential/sequencerStaking.cancel.test.ts:131]: same, plz if you duplicate some code, refactor it first.

File: test/sequential/sequencerStaking.cancel.test.ts [hunk 4]
```diff
@@ -0,0 +1,403 @@
...
+    await waitForNBlocks(10);
+  }
+});
+
+describe("Update cancellation -", () => {
+  let chain: any;
+  let testUser1: User;
+  let testUser2: User;
+  let testUser2Address: string;
+  beforeEach(async () => {
```
Comment (goncer) [inline test/sequential/sequencerStaking.cancel.test.ts:157]: I think this is also duplicated

File: test/sequential/sequencerStaking.cancel.test.ts [hunk 5]
```diff
@@ -0,0 +1,403 @@
...
+          .build(),
+      ),
+    );
+    await waitSudoOperationSuccess(cancelResolutionEvents, "SudoAsDone");
+    await testUser2.refreshAmounts(AssetWallet.BEFORE);
+    await testUser1.refreshAmounts(AssetWallet.BEFORE);
+    await waitForNBlocks((await Rolldown.disputePeriodLength()).toNumber());
+    await testUser2.refreshAmounts(AssetWallet.AFTER);
+    await testUser1.refreshAmounts(AssetWallet.AFTER);
+    const slashRewardCanceler = testUser2
```
Comment (goncer) [inline test/sequential/sequencerStaking.cancel.test.ts:216]: plz rename to something meaningful on this context

File: test/sequential/sequencerStaking.cancel.test.ts [hunk 6]
```diff
@@ -0,0 +1,403 @@
...
+      .getAsset(MGA_ASSET_ID)
+      ?.amountAfter.free!.sub(
+        testUser2.getAsset(MGA_ASSET_ID)?.amountBefore.free!,
+      );
+    const slashFineUpdater = testUser1
```
Comment (goncer) [inline test/sequential/sequencerStaking.cancel.test.ts:221]: plz rename to something meaningful on this context

File: test/sequential/sequencerStaking.cancel.test.ts [hunk 7]
```diff
@@ -0,0 +1,403 @@
...
+      .getAsset(MGA_ASSET_ID)
+      ?.amountBefore.reserved!.sub(
+        testUser1.getAsset(MGA_ASSET_ID)?.amountAfter.reserved!,
+      );
+    expect(slashRewardCanceler).bnGt(BN_ZERO);
+    expect(slashRewardCanceler).bnLt(await SequencerStaking.slashFineAmount());
+    expect(slashFineUpdater).bnEqual(await SequencerStaking.slashFineAmount());
```
Comment (goncer) [inline test/sequential/sequencerStaking.cancel.test.ts:228]: The amount the canceler gets is 20% of the slash fine amount I think. 
image
Plz adjust to be more precise

**Review by goncer — CHANGES_REQUESTED** [review]

File: test/sequential/sequencerStaking.cancel.test.ts [hunk 8]
```diff
@@ -0,0 +1,403 @@
...
+import { Sudo } from "../../utils/sudo";
+import {
+  waitNewBlock,
+  waitSudoOperationSuccess,
+} from "../../utils/eventListeners";
+import { Assets } from "../../utils/Assets";
+import { BN_ZERO } from "@polkadot/util";
+import { MGA_ASSET_ID } from "../../utils/Constants";
+
+async function waitSequencerAsSelected(seq: User, n: number = 80) {
```
[the remark this answers is not in the document]
Comment (AlexChetverov) [reply test/sequential/sequencerStaking.cancel.test.ts:25]: It was moved to Rolldown.ts

File: test/sequential/sequencerStaking.cancel.test.ts [hunk 9]
```diff
@@ -0,0 +1,403 @@
...
+  while (selectedSequencer.Ethereum !== seq.keyRingPair.address && n > 0) {
+    await waitNewBlock();
+    selectedSequencer = (
+      await api.query.sequencerStaking.selectedSequencer()
+    ).toHuman();
+    n--;
+  }
+}
+
+async function createACollatorUser() {
```
[the remark this answers is not in the document]
Comment (AlexChetverov) [reply test/sequential/sequencerStaking.cancel.test.ts:40]: Renamed

File: test/sequential/sequencerStaking.cancel.test.ts (hunk 1 above)
Comment (AlexChetverov) [reply test/sequential/sequencerStaking.cancel.test.ts:46]: It was moved to Rolldown.ts

File: test/sequential/sequencerStaking.cancel.test.ts (hunk 2 above)
Comment (AlexChetverov) [reply test/sequential/sequencerStaking.cancel.test.ts:72]: It was moved to Rolldown.ts

File: test/sequential/sequencerStaking.cancel.test.ts (hunk 3 above)
Comment (AlexChetverov) [reply test/sequential/sequencerStaking.cancel.test.ts:131]: Refactored

File: test/sequential/sequencerStaking.cancel.test.ts (hunk 4 above)
Comment (AlexChetverov) [reply test/sequential/sequencerStaking.cancel.test.ts:157]: Refactored

File: test/sequential/sequencerStaking.cancel.test.ts (hunk 7 above)
Comment (AlexChetverov) [reply test/sequential/sequencerStaking.cancel.test.ts:228]: Done

File: test/sequential/sequencerStaking.cancel.test.ts (hunk 6 above)
Comment (AlexChetverov) [reply test/sequential/sequencerStaking.cancel.test.ts]: renamed

File: test/sequential/sequencerStaking.cancel.test.ts [hunk 10]
```diff
@@ -1,403 +1,281 @@
...
   for (const chain in activeSequencers.toHuman()) {
     for (const seq of activeSequencers.toHuman()[chain] as string[]) {
-      if (
-        seq !== preSetupSequencers.Ethereum &&
-        seq !== preSetupSequencers.Arbitrum
-      ) {
-        await leaveSequencingIfAlreadySequencer(seq);
-        anySequencerGone = true;
+      if (seq !== null) {
+        await leaveSequencing(seq);
```
Comment (goncer) [inline test/sequential/sequencerStaking.cancel.test.ts:49]: is there any reason why you are ejecting the preSetupSequencers?

File: test/sequential/sequencerStaking.cancel.test.ts [hunk 11]
```diff
@@ -1,403 +1,281 @@
...
+    Sudo.sudoAs(testUser2, providingExtrinsic),
+  );
+  const sequencers = await SequencerStaking.activeSequencers();
+  expect(sequencers.toHuman().Ethereum).toContain(
+    testUser1.keyRingPair.address,
+  );
+  expect(sequencers.toHuman().Ethereum).toContain(
+    testUser2.keyRingPair.address,
+  );
+  testUser2Address = testUser2.ethAddress.toString();
```
Comment (goncer) [inline test/sequential/sequencerStaking.cancel.test.ts:70]: use the keyring. address instead, I run into soo many problems with that ethAddress. I am about to remove it soon

File: test/sequential/sequencerStaking.cancel.test.ts [hunk 12]
```diff
@@ -1,403 +1,281 @@
...
+        .build(),
+    ),
+  );
+  await waitSudoOperationSuccess(cancelResolutionEvents, "SudoAsDone");
+  await waitForNBlocks(disputePeriodLength);
+  updaterRightsStatus = await Rolldown.sequencerRights(
+    chain,
+    testUser1.keyRingPair.address,
+  );
+  expect(updaterRightsStatus.cancelRights.toString()).toBe("0");
```
Comment (goncer) [inline test/sequential/sequencerStaking.cancel.test.ts:200]: plz additionally validate that ,  
- it does no longer belong to the active set,
- it does not have read rights anymore

File: test/sequential/sequencerStaking.cancel.test.ts [hunk 13]
```diff
@@ -1,403 +1,281 @@
...
+});
 
-  it("GIVEN a slashed sequencer, WHEN slashed it can not provide any update / cancel until the next session ( if gets elected )", async () => {
-    const { reqIdCanceled, api } = await createAnUpdateAndCancelIt(
-      testUser1,
+it("GIVEN a sequencer, WHEN <in-correctly> canceling an update AND some pending updates/cancels, THEN it can be still slashed and kicked, cancels & updates will be executed.", async () => {
```
Comment (goncer) [inline test/sequential/sequencerStaking.cancel.test.ts:203]: looks like the test is not doing what It should be doing. The test should be about: 

1. u1 - create an update
2. u2 - Cancel it
3. u2 - create an update
4. u1 - cancel it.
5. judge tells that u2 cancel is wrong ( at step 2, with justified false ) 
6. u2 it must be kicked.
7. update 1 is executed
8. judge tells that u1 cancel is wrong. 
9. u1 must be kicked.
10. even though u2 was kicked, the update is executed.
Expected: 
The most important is to ensure that, u2 got kicked and the update was executed.
* updates for u1 and u2 are executed. ( verify the execution or execution result )
* both updaters are kicked. ( no read, no writes )

**Review by goncer — CHANGES_REQUESTED** [review]

File: test/sequential/sequencerStaking.cancel.test.ts (hunk 10 above)
Comment (AlexChetverov) [reply test/sequential/sequencerStaking.cancel.test.ts:49]: This way I reduced the time it took to complete the tests.

File: test/sequential/sequencerStaking.cancel.test.ts (hunk 11 above)
Comment (AlexChetverov) [reply test/sequential/sequencerStaking.cancel.test.ts]: Done

File: test/sequential/sequencerStaking.cancel.test.ts (hunk 12 above)
Comment (AlexChetverov) [reply test/sequential/sequencerStaking.cancel.test.ts:206]: Done

File: test/sequential/sequencerStaking.cancel.test.ts (hunk 13 above)
Comment (AlexChetverov) [reply test/sequential/sequencerStaking.cancel.test.ts:210]: Test was changed

File: test/sequential/sequencerStaking.cancel.test.ts (hunk 12 above)
Comment (goncer) [reply test/sequential/sequencerStaking.cancel.test.ts:206]: ty!

File: test/sequential/sequencerStaking.cancel.test.ts [hunk 14]
```diff
@@ -197,71 +197,85 @@ it("GIVEN a slashed sequencer, WHEN slashed it can not provide any update / canc
...
+    )
+    .withDeposit(
+      txIndex + 1,
+      testUser2.keyRingPair.address,
+      testUser1.keyRingPair.address,
+      BN_MILLION,
+    )
+    .on(chain)
+    .build();
+  const { reqIdCanceled: reqIdCanceled2, api: api2 } =
```
Comment (goncer) [inline test/sequential/sequencerStaking.cancel.test.ts:243]: why api2? :O

File: test/sequential/sequencerStaking.cancel.test.ts [hunk 15]
```diff
@@ -197,71 +197,85 @@ it("GIVEN a slashed sequencer, WHEN slashed it can not provide any update / canc
...
-  const txIndex2 = await Rolldown.lastProcessedRequestOnL2(chain);
   await Rolldown.waitForReadRights(judge.keyRingPair.address);
-  cancelResolutionEvents = await Sudo.asSudoFinalized(
+  await testUser1.refreshAmounts(AssetWallet.BEFORE);
+  const cancelResolutionEvent2 = await Sudo.asSudoFinalized(
     Sudo.sudoAsWithAddressString(
       judge.keyRingPair.address,
       new L2Update(api2)
-        .withCancelResolution(txIndex2, reqIdCanceled2, false)
+        .withCancelResolution(txIndex + 1, reqIdCanceled2, false)
```
Comment (goncer) [inline test/sequential/sequencerStaking.cancel.test.ts:270]: here I think you can use the getLatestExecutedRequestId, but I think its ok with those txindex+1

File: test/sequential/sequencerStaking.cancel.test.ts [hunk 16]
```diff
@@ -197,71 +197,85 @@ it("GIVEN a slashed sequencer, WHEN slashed it can not provide any update / canc
...
         .on("Ethereum")
```
Comment (goncer) [inline test/sequential/sequencerStaking.cancel.test.ts:271]: use the chain constant u have plz

File: test/sequential/sequencerStaking.cancel.test.ts [hunk 17]
```diff
@@ -272,6 +286,13 @@ it("GIVEN a sequencer, WHEN <in-correctly> canceling an update AND some pending
...
+  const sequencers = await SequencerStaking.activeSequencers();
+  expect(sequencers.toHuman().Ethereum).not.toContain(
+    testUser1.keyRingPair.address,
+  );
+  expect(sequencers.toHuman().Ethereum).not.toContain(
+    testUser2.keyRingPair.address,
+  );
   expect(testUser1PenaltyValue).bnEqual(
     await SequencerStaking.slashFineAmount(),
   );
```
Comment (goncer) [inline test/sequential/sequencerStaking.cancel.test.ts:298]: It looks superb, but it seems you are missing the checks for the processed requests.  Both updates with the deposits must be executed, and I can not find any place where u are validating those executions. 
There must be some "RequestExecutedEvents" or perhaps you can validate the balances for the user used in the deposits.

Comment (goncer) [reply test/sequential/sequencerStaking.cancel.test.ts:298]: here I can see that 
https://github.com/mangata-finance/mangata-e2e/pull/463/files/96ac491ccd2882b33c34fbde98cbbf113023ca9a..46d0f3882783ddfbd4d2d56f4183b6d53258b615#diff-251c1940a8a54baa9fc6c3c0bb65212a09822b46d7ab4a6d214ef14524f288d4R355
after the first deposit, the user must have got some tokens with the same address as his address ( the sequencer address who provided the update ) , and looks that the same must have happened for the second ones,

**Review by goncer — CHANGES_REQUESTED** [review]

File: test/sequential/sequencerStaking.cancel.test.ts (hunk 14 above)
Comment (AlexChetverov) [reply test/sequential/sequencerStaking.cancel.test.ts:243]: Fixed

File: test/sequential/sequencerStaking.cancel.test.ts (hunk 15 above)
Comment (AlexChetverov) [reply test/sequential/sequencerStaking.cancel.test.ts:270]: I changed it

File: test/sequential/sequencerStaking.cancel.test.ts (hunk 16 above)
Comment (AlexChetverov) [reply test/sequential/sequencerStaking.cancel.test.ts:271]: Done

File: test/sequential/sequencerStaking.cancel.test.ts [hunk 18]
```diff
@@ -300,3 +299,149 @@ it("GIVEN a sequencer, WHEN <in-correctly> canceling an update AND some pending
...
+    ?.amountBefore.reserved!.sub(
+      testUser2.getAsset(GASP_ASSET_ID)?.amountAfter.reserved!,
+    );
+  expect(sequencers.toHuman().Ethereum).toContain(testUser2Address);
+  expect(testUser2PenaltyValue).bnEqual(
+    await SequencerStaking.slashFineAmount(),
+  );
+});
+
+it("GIVEN a sequencer, WHEN <in-correctly> canceling an update AND some pending updates/cancels AND users stakes new token THEN they return sequencersRights", async () => {
```
Comment (goncer) [inline test/sequential/sequencerStaking.cancel.test.ts:344]: same as before, can not understand what is this test about

File: test/sequential/sequencerStaking.cancel.test.ts [hunk 19]
```diff
@@ -300,3 +299,149 @@ it("GIVEN a sequencer, WHEN <in-correctly> canceling an update AND some pending
...
+  const { reqIdCanceled: reqIdCanceled2 } = await createAnUpdateAndCancelIt(
+    testUser2,
+    testUser1.keyRingPair.address,
+    chain,
+    updateValue,
+  );
+  await waitForNBlocks(disputePeriodLength);
+  providingExtrinsic = await SequencerStaking.provideSequencerStaking(
+    (await SequencerStaking.minimalStakeAmount()).muln(2),
+    "Ethereum",
```
Comment (goncer) [inline test/sequential/sequencerStaking.cancel.test.ts:387]: use the chain constant plz

File: test/sequential/sequencerStaking.cancel.test.ts [hunk 20]
```diff
@@ -300,3 +299,149 @@ it("GIVEN a sequencer, WHEN <in-correctly> canceling an update AND some pending
...
+  );
+  const cancelResolutionEvent1 = await Sudo.batchAsSudoFinalized(
+    Sudo.sudoAsWithAddressString(
+      judge.keyRingPair.address,
+      new L2Update(api)
+        .withCancelResolution(txIndex1, reqIdCanceled1, false)
+        .on("Ethereum")
```
Comment (goncer) [inline test/sequential/sequencerStaking.cancel.test.ts:394]: use the chain constant

File: test/sequential/sequencerStaking.cancel.test.ts [hunk 21]
```diff
@@ -300,3 +299,149 @@ it("GIVEN a sequencer, WHEN <in-correctly> canceling an update AND some pending
...
+    await SequencerStaking.slashFineAmount(),
+  );
+  expect(testUser2PenaltyValue).bnEqual(
+    await SequencerStaking.slashFineAmount(),
+  );
+
+  expect(testUser1RightsStatus.readRights.toString()).toBe("1");
+  expect(testUser2RightsStatus.readRights.toString()).toBe("1");
+  expect(testUser2RightsStatus.cancelRights.toString()).toBe("2");
+  expect(testUser1RightsStatus.cancelRights.toString()).toBe("2");
```
Comment (goncer) [inline test/sequential/sequencerStaking.cancel.test.ts:446]: :clap:

File: test/sequential/sequencerStaking.cancel.test.ts [hunk 22]
```diff
@@ -300,3 +299,149 @@ it("GIVEN a sequencer, WHEN <in-correctly> canceling an update AND some pending
...
+    .on(chain)
+    .build();
+  const { reqIdCanceled: reqIdCanceled2 } = await createAnUpdateAndCancelIt(
+    testUser2,
+    testUser1.keyRingPair.address,
+    chain,
+    updateValue,
+  );
+  await waitForNBlocks(disputePeriodLength);
+  providingExtrinsic = await SequencerStaking.provideSequencerStaking(
```
Comment (goncer) [inline test/sequential/sequencerStaking.cancel.test.ts:385]: I've seen that you are reusing this global variable. 
Plz use local constants for local context, and avoid using global variables for local context stuff.

**Review by goncer — CHANGES_REQUESTED** [review]

File: test/sequential/sequencerStaking.cancel.test.ts [hunk 23]
```diff
@@ -300,3 +299,149 @@ it("GIVEN a sequencer, WHEN <in-correctly> canceling an update AND some pending
     await SequencerStaking.slashFineAmount(),
   );
```
[the remark this answers is not in the document]
Comment (AlexChetverov) [reply test/sequential/sequencerStaking.cancel.test.ts:300]: We do not fulfill the deposit even if the cancellation was made in error.

File: test/sequential/sequencerStaking.cancel.test.ts [hunk 24]
```diff
@@ -300,3 +299,149 @@ it("GIVEN a sequencer, WHEN <in-correctly> canceling an update AND some pending
...
+  await testUser1.refreshAmounts(AssetWallet.BEFORE);
+  const { reqIdCanceled, api } = await createAnUpdateAndCancelIt(
+    testUser1,
+    testUser2Address,
+    chain,
+  );
+  //await waitForNBlocks(disputePeriodLength);
+  await Rolldown.waitForReadRights(testUser2Address);
+  const txIndex = await Rolldown.lastProcessedRequestOnL2(chain);
+  //the cancellation is incorrectly
```
[the remark this answers is not in the document]
Comment (AlexChetverov) [reply test/sequential/sequencerStaking.cancel.test.ts:313]: The comment was moved

File: test/sequential/sequencerStaking.cancel.test.ts [hunk 25]
```diff
@@ -208,52 +208,52 @@ it("GIVEN a slashed sequencer, WHEN slashed it can not provide any update / canc
...
   const [judge] = setupUsers();
   await Sudo.batchAsSudoFinalized(
     Assets.mintNative(judge),
     Sudo.sudoAs(judge, providingExtrinsic),
   );
 
   const {
-    txIndex,
-    api: api1,
+    api,
```
[the remark this answers is not in the document]
Comment (AlexChetverov) [reply test/sequential/sequencerStaking.cancel.test.ts:218]: delete and create global variable

File: test/sequential/sequencerStaking.cancel.test.ts [hunk 26]
```diff
@@ -208,52 +208,52 @@ it("GIVEN a slashed sequencer, WHEN slashed it can not provide any update / canc
...
   await waitForNBlocks(disputePeriodLength);
   await testUser2.refreshAmounts(AssetWallet.BEFORE);
   const cancelResolutionEvent1 = await Sudo.asSudoFinalized(
     Sudo.sudoAsWithAddressString(
       judge.keyRingPair.address,
-      new L2Update(api1)
-        .withCancelResolution(txIndex, reqIdCanceled1, false)
+      new L2Update(api)
+        .withCancelResolution(txIndex1, reqIdCanceled1, false)
         .on("Ethereum")
```
[the remark this answers is not in the document]
Comment (AlexChetverov) [reply test/sequential/sequencerStaking.cancel.test.ts:257]: done

File: test/sequential/sequencerStaking.cancel.test.ts [hunk 27]
```diff
@@ -300,3 +299,149 @@ it("GIVEN a sequencer, WHEN <in-correctly> canceling an update AND some pending
     await SequencerStaking.slashFineAmount(),
   );
 });
+
+it("GIVEN a sequencer, WHEN <in-correctly> canceling an update AND cancelator stakes new token THEN it will be slashed and returned to the sequencer pool", async () => {
```
[the remark this answers is not in the document]
Comment (AlexChetverov) [reply test/sequential/sequencerStaking.cancel.test.ts:303]: Rename, pls check

File: test/sequential/sequencerStaking.cancel.test.ts (hunk 18 above)
Comment (AlexChetverov) [reply test/sequential/sequencerStaking.cancel.test.ts:344]: Rename, pls check

File: test/sequential/sequencerStaking.cancel.test.ts (hunk 19 above)
Comment (AlexChetverov) [reply test/sequential/sequencerStaking.cancel.test.ts:387]: done

File: test/sequential/sequencerStaking.cancel.test.ts (hunk 20 above)
Comment (AlexChetverov) [reply test/sequential/sequencerStaking.cancel.test.ts:394]: done

File: test/sequential/sequencerStaking.cancel.test.ts (hunk 22 above)
Comment (AlexChetverov) [reply test/sequential/sequencerStaking.cancel.test.ts:385]: Done

File: test/sequential/sequencerStaking.cancel.test.ts [hunk 28]
```diff
@@ -300,3 +299,149 @@ it("GIVEN a sequencer, WHEN <in-correctly> canceling an update AND some pending
...
+  providingExtrinsic = await SequencerStaking.provideSequencerStaking(
+    (await SequencerStaking.minimalStakeAmount()).muln(2),
+    "Ethereum",
+  );
+  const cancelResolutionEvents = await Sudo.batchAsSudoFinalized(
+    Sudo.sudoAsWithAddressString(
+      testUser2Address,
+      new L2Update(api)
+        .withCancelResolution(txIndex, reqIdCanceled, false)
+        .on("Ethereum")
```
[the remark this answers is not in the document]
Comment (AlexChetverov) [reply test/sequential/sequencerStaking.cancel.test.ts:323]: done

File: utils/rollDown/Rolldown.ts [hunk 29]
```diff
@@ -352,20 +394,24 @@ export async function createAnUpdate(
...
   let reqId = 0;
+  let executionBlockNumber: any;
   await Sudo.asSudoFinalized(
     Sudo.sudoAsWithAddressString(address, update),
   ).then(async (events) => {
+    executionBlockNumber = findEventData(events, "rolldown.L1ReadStored")[0][2]
+      .toString()
+      .replaceAll(",", "");
     await waitSudoOperationSuccess(events, "SudoAsDone");
     reqId = Rolldown.getRequestIdFromEvents(events);
```
Comment (goncer) [inline utils/rollDown/Rolldown.ts:412]: is not the same ? execution block number and reqId ?

**Review by goncer — APPROVED** [review]

**Pull request merged**